webapp/tests/karma/ts/services/enketo.service.spec.ts

```diff
@@ -1470,4 +1475,51 @@ describe('Enketo service', () => {
     });
   });
 
+  describe('renderContactForm', () => {
+    beforeEach(() => {
+      service.setFormTitle = sinon.stub();
+      dbGetAttachment.resolves('<form/>');
+      translateService.get.callsFake((key) => `translated key ${key}`);
+    });
+
+    const callbackMock = () => {};
+    const instanceData = {
+      health_center: {
+        type: 'contact',
+        contact_type: 'health_center',
+        parent: 'parent',
+      },
+    };
+    const formDoc = {
+      ...mockEnketoDoc('myform'),
+      title: 'New Area',
+    };
+
+    it('should translate titleKey when provided', async () => {
+      await service.renderContactForm({
+        selector: $('<div></div>'),
+        formDoc,
+        instanceData,
+        editedListener: callbackMock,
+        valuechangeListener: callbackMock,
+        titleKey: 'contact.type.health_center.new',
+      });
+
+      expect(service.setFormTitle.callCount).to.be.equal(1);
+      expect(service.setFormTitle.args[0][1]).to.be.equal('translated key contact.type.health_center.new');
+    });
+
+    it('should fallback to translate document title when the titleKey is not available', async () => {
+      await service.renderContactForm({
+        selector: $('<div></div>'),
+        formDoc,
+        instanceData,
+        editedListener: callbackMock,
+        valuechangeListener: callbackMock,
+      });
+
+      expect(service.setFormTitle.callCount).to.be.equal(1);
+      expect(service.setFormTitle.args[0][1]).to.be.equal('translated');
```

Comment:
I think it would be helpful for the reader if you configure the returned value of `TranslateFrom` service and use a more suggestive string than "translated".